// c2c-reviews/routes.js
var fs = require('fs');

module.exports = function($stateProvider) {
  $stateProvider
    .state('reviews', {
      url: '/reviews',
      abstract: true,
      template: '<div ui-view></div>'
    })
    .state('reviews.list', {
      url: '/?submission',
      controller: require('./controllers/list'),
      template: fs.readFileSync(__dirname + '/templates/list.html'),
      resolve: { 
        reviews: require('./resolve/reviews'), 
        submission: require('./resolve/submission'),
      }
    })
    .state('reviews.new', {
      url: '/new?submission',
      controller: require('./controllers/new'),
      template: fs.readFileSync(__dirname + '/templates/form.html'),
      resolve: { 
        reviews: require('./resolve/reviews'),
        submission: require('./resolve/submission') 
      }
    })
    .state('reviews.show', {
      url: '/:id',
      controller: require('./controllers/show'),
      template: fs.readFileSync(__dirname + '/templates/show.html'),
      resolve: { review: require('./resolve/review') }
    })
    .state('reviews.edit', {
      url: '/:id/edit',
      controller: require('./controllers/edit'),
      template: fs.readFileSync(__dirname + '/templates/form.html'),
      resolve: { submission: require('./resolve/submission') }
    });
}
