// c2c-submissions/routes.js
var fs = require('fs');

module.exports = function($stateProvider) {
  $stateProvider
    .state('submissions', {
      url: '/submissions',
      abstract: true,
      template: '<div ui-view></div>'
    })
    .state('submissions.list', {
      url: '/?challenge',
      controller: require('./controllers/list'),
      template: fs.readFileSync(__dirname + '/templates/list.html'),
      resolve: { 
        challenge: require('./resolve/challenge')
      }
    })
    .state('submissions.new', {
      url: '/new?challenge',
      controller: require('./controllers/new'),
      template: fs.readFileSync(__dirname + '/templates/form.html'),
      resolve: { 
        submissions: require('./resolve/submissions'),
        challenge: require('./resolve/challenge') 
      }
    })
    .state('submissions.show', {
      url: '/:id',
      controller: require('./controllers/show'),
      template: fs.readFileSync(__dirname + '/templates/show.html'),
      resolve: { submission: require('./resolve/submission') }
    })
    .state('submissions.edit', {
      url: '/:id/edit',
      controller: require('./controllers/edit'),
      template: fs.readFileSync(__dirname + '/templates/form.html'),
      resolve: { submission: require('./resolve/submission') }
    });
}
